sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/trees/TreeNodeSuite.scala

```diff
@@ -261,4 +283,146 @@ class TreeNodeSuite extends SparkFunSuite {
       assert(actual === expected)
     }
   }
+
+  test("toJSON") {
+    def expected(json: String): String = {
```

Comment:
how about
```
def checkJsonString(value: Any, json: String): Unit = {
  compareJSON(
    JsonTestTreeNode(value).toJSON,
    s"""[{"class":"${classOf[JsonTestTreeNode].getName}","num-children":0,"arg":$json}]"""
  )
}
```
then we can simply write `checkJsonString(true, "true")` instead of `compareJSON(JsonTestTreeNode(true).toJSON, expected("true"))`